src/main/java/org/myrobotlab/service/LcdDisplay.java

```diff
@@ -0,0 +1,232 @@
+  private static final long serialVersionUID = 1L;
+
+  private Pcf8574 i2cControler;
+
+  private final byte LCD_CLEARDISPLAY = (byte) 0x01;
+  private final byte LCD_RETURNHOME = (byte) 0x02;
+  private final byte LCD_ENTRYMODESET = (byte) 0x04;
+  private final byte LCD_DISPLAYCONTROL = (byte) 0x08;
+  private final byte LCD_CURSORSHIFT = (byte) 0x10;
+  private final byte LCD_FUNCTIONSET = (byte) 0x20;
+  private final byte LCD_SETCGRAMADDR = (byte) 0x40;
+  private final byte LCD_SETDDRAMADDR = (byte) 0x80;
+
+  // flags for display entry mode 
+  private final byte LCD_ENTRYRIGHT = (byte) 0x00;
+  private final byte LCD_ENTRYLEFT = (byte) 0x02;
+  private final byte LCD_ENTRYSHIFTINCREMENT = (byte) 0x01;
+  private final byte LCD_ENTRYSHIFTDECREMENT = (byte) 0x00;
+
+  // flags for display on/off control
+  private final byte LCD_DISPLAYON = (byte) 0x04;
+  private final byte LCD_DISPLAYOFF = (byte) 0x00;
+  private final byte LCD_CURSORON = (byte) 0x02;
+  private final byte LCD_CURSOROFF = (byte) 0x00;
+  private final byte LCD_BLINKON = (byte) 0x01;
+  private final byte LCD_BLINKOFF = (byte) 0x00;
+
+  // flags for display/cursor shift
+  private final byte LCD_DISPLAYMOVE = (byte) 0x08;
+  private final byte LCD_CURSORMOVE = (byte) 0x00;
+  private final byte LCD_MOVERIGHT = (byte) 0x04;
+  private final byte LCD_MOVELEFT = (byte) 0x00;
+
+  // flags for function set
+  private final byte LCD_8BITMODE = (byte) 0x10;
+  private final byte LCD_4BITMODE = (byte) 0x00;
+  private final byte LCD_2LINE = (byte) 0x08;
+  private final byte LCD_1LINE = (byte) 0x00;
+  private final byte LCD_5x10DOTS = (byte) 0x04;
+  private final byte LCD_5x8DOTS = (byte) 0x00;
+
+  // flags for backlight control
+  private final byte LCD_BACKLIGHT = (byte) 0x08;
+  private final byte LCD_NOBACKLIGHT = (byte) 0x00;
+
+  private final byte En = (byte) 0b00000100; // Enable bit
+  private final byte Rw = (byte) 0b00000010; // Read/Write bit
+  private final byte Rs = (byte) 0b00000001; // Register select bit
+
+  private boolean backLight;
+  public Map<Integer, String> screenContent = new HashMap<Integer, String>();
+
+  public void attach(Pcf8574 pcf8574) {
+    // we need more checkup here...
+    if (pcf8574.isAttached) {
+      this.i2cControler = pcf8574;
+      setReady(true);
+      log.info("{} attach {}", getName(), pcf8574.getName());
+    } else {
+      log.error("Cannot attach {} to {}", getName(), pcf8574.getName());
+    }
+  }
+
+  /*
+   * INIT LCD panel
+   */
+  public void init() {
+    log.info("Init I2C Display");
+    lcd_write((byte) 0x03);
+    lcd_write((byte) 0x03);
+    lcd_write((byte) 0x03);
+    lcd_write((byte) 0x02);
+    lcd_write((byte) (LCD_FUNCTIONSET | LCD_2LINE | LCD_5x8DOTS | LCD_4BITMODE));
+    lcd_write((byte) (LCD_DISPLAYCONTROL | LCD_DISPLAYON));
+    lcd_write((byte) (LCD_CLEARDISPLAY));
+    lcd_write((byte) (LCD_ENTRYMODESET | LCD_ENTRYLEFT));
+  }
+
+  /*
+   * Send byte to PCF controller

```

Comment:
All these comments are awesome ..
But they should be JavaDoc .. e.g.
if you do this -> /**
and hit return, eclipse auto-creates the rest of the javadoc - you just fill it in

Reply:
/**
   *  checked in
   * 
   */